manifests/params.pp

```diff
@@ -156,6 +156,8 @@
               $service_name  = 'ntpd'
               $keys_file     = '/etc/ntp.keys'
               $package_name  = $default_package_name
+              #SLES12 uses sytemd as its service provider, hence why we hardcode it here. 
+              $service_provider = 'systemd'
```

Comment:
Personal preference: I'd prefer the `$service_provider` to be set more explicitly... so it should be set to `undef` on all of the other cases. Similar to how `$maxpoll` is handled. It's pretty obvious to the majority of people reading the code that `$service_provider` should be `undef` in all platforms != SLES12, but it doesn't hurt to be explicit IMO.